lambda layer > 60 seem to break with "Error: Cannot find module './async'"

### Is there an existing issue for this?

- [X] I have checked for existing issues https://github.com/getsentry/sentry-javascript/issues
- [X] I have reviewed the documentation https://docs.sentry.io/
- [X] I am using the latest SDK release https://github.com/getsentry/sentry-javascript/releases

### How do you use Sentry?

Sentry Saas (sentry.io)

### Which package are you using?

@sentry/serverless

### SDK Version

6.19.4

### Framework Version

N/A

### Link to Sentry event

_No response_

### Steps to Reproduce

Cloudformation:

```yaml
Globals:
  Function:
    Layers:
      - !Sub 'arn:aws:lambda:${AWS::Region}:943013980633:layer:SentryNodeServerlessSDK:61'
```

In lambda:

```js
require('@sentry/serverless');
```

I am in `ap-southeast-1` if that matters. Basically 61, 62 fail on an async import. 63 fails on some other import.

### Expected Result

Is functional.

### Actual Result

```
{
  "errorType": "Runtime.ImportModuleError",
  "errorMessage": "Error: Cannot find module './async'\nRequire stack:\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/utils/build/dist/index.js\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/hub/build/dist/scope.js\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/hub/build/dist/index.js\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/minimal/build/dist/index.js\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/core/build/dist/index.js\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/node/build/dist/index.js\n- /opt/nodejs/node_modules/@sentry/serverless/build/dist/awslambda.js\n- /opt/nodejs/node_modules/@sentry/serverless/build/dist/index.js\n- /var/task/onconnect.js\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js",
  "trace": [
    "Runtime.ImportModuleError: Error: Cannot find module './async'",
    "Require stack:",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/utils/build/dist/index.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/hub/build/dist/scope.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/hub/build/dist/index.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/minimal/build/dist/index.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/core/build/dist/index.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/node/build/dist/index.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/build/dist/awslambda.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/build/dist/index.js",
    "- /var/task/onconnect.js",
    "- /var/runtime/UserFunction.js",
    "- /var/runtime/index.js",
    "    at _loadUserApp (/var/runtime/UserFunction.js:202:13)",
    "    at Object.module.exports.load (/var/runtime/UserFunction.js:242:17)",
    "    at Object.<anonymous> (/var/runtime/index.js:43:30)",
    "    at Module._compile (internal/modules/cjs/loader.js:1085:14)",
    "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)",
    "    at Module.load (internal/modules/cjs/loader.js:950:32)",
    "    at Function.Module._load (internal/modules/cjs/loader.js:790:12)",
    "    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:75:12)",
    "    at internal/main/run_main_module.js:17:47"
  ]
}
```

62

```js
{
  "errorType": "Runtime.ImportModuleError",
  "errorMessage": "Error: Cannot find module './async'\nRequire stack:\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/utils/build/dist/index.js\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/hub/build/dist/scope.js\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/hub/build/dist/index.js\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/minimal/build/dist/index.js\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/core/build/dist/index.js\n- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/node/build/dist/index.js\n- /opt/nodejs/node_modules/@sentry/serverless/build/dist/awslambda.js\n- /opt/nodejs/node_modules/@sentry/serverless/build/dist/index.js\n- /var/task/ping.js\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js",
  "trace": [
    "Runtime.ImportModuleError: Error: Cannot find module './async'",
    "Require stack:",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/utils/build/dist/index.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/hub/build/dist/scope.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/hub/build/dist/index.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/minimal/build/dist/index.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/core/build/dist/index.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/node/build/dist/index.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/build/dist/awslambda.js",
    "- /opt/nodejs/node_modules/@sentry/serverless/build/dist/index.js",
    "- /var/task/ping.js",
    "- /var/runtime/UserFunction.js",
    "- /var/runtime/index.js",
    "    at _loadUserApp (/var/runtime/UserFunction.js:202:13)",
    "    at Object.module.exports.load (/var/runtime/UserFunction.js:242:17)",
    "    at Object.<anonymous> (/var/runtime/index.js:43:30)",
    "    at Module._compile (internal/modules/cjs/loader.js:1085:14)",
    "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)",
    "    at Module.load (internal/modules/cjs/loader.js:950:32)",
    "    at Function.Module._load (internal/modules/cjs/loader.js:790:12)",
    "    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:75:12)",
    "    at internal/main/run_main_module.js:17:47"
  ]
}
```

63

```js
{
  "errorType": "Runtime.ImportModuleError",
  "errorMessage": "Error: Cannot find module '/opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/tracing/build/npm/cjs/index.js'. Please verify that the package.json has a valid \"main\" entry",
  "trace": [
    "Runtime.ImportModuleError: Error: Cannot find module '/opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/tracing/build/npm/cjs/index.js'. Please verify that the package.json has a valid \"main\" entry",
    "    at _loadUserApp (/var/runtime/UserFunction.js:202:13)",
    "    at Object.module.exports.load (/var/runtime/UserFunction.js:242:17)",
    "    at Object.<anonymous> (/var/runtime/index.js:43:30)",
    "    at Module._compile (internal/modules/cjs/loader.js:1085:14)",
    "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)",
    "    at Module.load (internal/modules/cjs/loader.js:950:32)",
    "    at Function.Module._load (internal/modules/cjs/loader.js:790:12)",
    "    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:75:12)",
    "    at internal/main/run_main_module.js:17:47"
  ]
}
```
64

```js
{
  "errorType": "Runtime.ImportModuleError",
  "errorMessage": "Error: Cannot find module '/opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/tracing/build/npm/cjs/index.js'. Please verify that the package.json has a valid \"main\" entry",
  "trace": [
    "Runtime.ImportModuleError: Error: Cannot find module '/opt/nodejs/node_modules/@sentry/serverless/node_modules/@sentry/tracing/build/npm/cjs/index.js'. Please verify that the package.json has a valid \"main\" entry",
    "    at _loadUserApp (/var/runtime/UserFunction.js:202:13)",
    "    at Object.module.exports.load (/var/runtime/UserFunction.js:242:17)",
    "    at Object.<anonymous> (/var/runtime/index.js:43:30)",
    "    at Module._compile (internal/modules/cjs/loader.js:1085:14)",
    "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)",
    "    at Module.load (internal/modules/cjs/loader.js:950:32)",
    "    at Function.Module._load (internal/modules/cjs/loader.js:790:12)",
    "    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:75:12)",
    "    at internal/main/run_main_module.js:17:47"
  ]
}
```

index.js

```
const Sentry = require('@sentry/serverless');

Sentry.AWSLambda.init({
  sampleRate: 0,
});

async function main() {
  console.log(Sentry.VERSION)
  return { statusCode: 200, body: 'pong' };
}

exports.handler = Sentry.AWSLambda.wrapHandler(main, {
  ignoreSentryErrors: true,
});
```

package.json

```
{
  "name": "ws-ping",
  "version": "1.0.0",
  "main": "ping.js",
  "devDependencies": {
    "@sentry/serverless": "6.19.6",
    "eslint": "8.13.0",
    "eslint-config-airbnb-base": "15.0.0",
    "eslint-plugin-import": "2.26.0"
  },
  "dependencies": {},
  "scripts": {
    "test:lint": "eslint ./"
  }
}
```

While you're working on V7 maybe you guys want to remove 62 from https://docs.sentry.io/platforms/node/guides/aws-lambda/layer/ if it directly causes issues.

--- a/packages/minimal/.npmignore
+++ b/packages/minimal/.npmignore
@@ -1,0 +1,11 @@
+# The paths in this file are specified so that they align with the file structure in `./build` after this file is copied
+# into it by the prepack script `scripts/prepack.ts`.
+
+*
+
+# TODO remove bundles (which in the tarball are inside `build`) in v7
+!/build/**/*
+
+!/dist/**/*
+!/esm/**/*
+!/types/**/*

--- a/packages/node/.npmignore
+++ b/packages/node/.npmignore
@@ -1,0 +1,11 @@
+# The paths in this file are specified so that they align with the file structure in `./build` after this file is copied
+# into it by the prepack script `scripts/prepack.ts`.
+
+*
+
+# TODO remove bundles (which in the tarball are inside `build`) in v7
+!/build/**/*
+
+!/dist/**/*
+!/esm/**/*
+!/types/**/*

--- a/packages/serverless/scripts/build-awslambda-layer.js
+++ b/packages/serverless/scripts/build-awslambda-layer.js
@@ -145,6 +145,13 @@ async function main() {
   const zipFilename = `sentry-node-serverless-${version}.zip`;
 
   try {
+    fs.symlinkSync(path.resolve(destRoot, 'build', 'dist'), path.resolve(destRoot, 'dist'));
+    fs.symlinkSync(path.resolve(destRoot, 'build', 'esm'), path.resolve(destRoot, 'esm'));
+  } catch (error) {
+    console.error(error);
+  }
+
+  try {
     fs.unlinkSync(path.resolve(dist, zipFilename));
   } catch (error) {
     // If the ZIP file hasn't been previously created (e.g. running this script for the first time),

--- a/packages/tracing/.npmignore
+++ b/packages/tracing/.npmignore
@@ -1,0 +1,11 @@
+# The paths in this file are specified so that they align with the file structure in `./build` after this file is copied
+# into it by the prepack script `scripts/prepack.ts`.
+
+*
+
+# TODO remove bundles (which in the tarball are inside `build`) in v7
+!/build/**/*
+
+!/dist/**/*
+!/esm/**/*
+!/types/**/*

--- a/packages/tracing/package.json
+++ b/packages/tracing/package.json
@@ -29,7 +29,7 @@
     "jsdom": "^16.2.2"
   },
   "scripts": {
-    "build": "run-p build:cjs build:esm build:types build:bundle && ts-node ../../scripts/prepack.ts #necessary for integration tests",
+    "build": "run-p build:cjs build:esm build:types build:bundle && ts-node ../../scripts/prepack.ts --bundles #necessary for integration tests",
     "build:bundle": "rollup --config",
     "build:cjs": "tsc -p tsconfig.cjs.json",
     "build:dev": "run-p build:cjs build:esm build:types",
@@ -82,8 +82,8 @@
   "sideEffects": [
     "./dist/index.js",
     "./esm/index.js",
-    "./npm/dist/index.js",
-    "./npm/esm/index.js",
+    "./build/npm/dist/index.js",
+    "./build/npm/esm/index.js",
     "./src/index.ts"
   ]
 }

--- a/packages/types/.npmignore
+++ b/packages/types/.npmignore
@@ -1,0 +1,11 @@
+# The paths in this file are specified so that they align with the file structure in `./build` after this file is copied
+# into it by the prepack script `scripts/prepack.ts`.
+
+*
+
+# TODO remove bundles (which in the tarball are inside `build`) in v7
+!/build/**/*
+
+!/dist/**/*
+!/esm/**/*
+!/types/**/*

--- a/packages/utils/.npmignore
+++ b/packages/utils/.npmignore
@@ -1,0 +1,11 @@
+# The paths in this file are specified so that they align with the file structure in `./build` after this file is copied
+# into it by the prepack script `scripts/prepack.ts`.
+
+*
+
+# TODO remove bundles (which in the tarball are inside `build`) in v7
+!/build/**/*
+
+!/dist/**/*
+!/esm/**/*
+!/types/**/*

